src/vscode-bicep/src/commands/deploy.ts

```diff
@@ -157,18 +158,36 @@ export class DeployCommand implements Command {
         }
       }
     } catch (err) {
-      this.outputChannelManager.appendToOutputChannel(
-        err instanceof UserCancelledError
-          ? `Deployment canceled for ${documentPath}.`
-          : `Deployment failed for ${documentPath}. ${parseError(err).message}`
-      );
+      let errorMessage: string;
+
+      if (err instanceof UserCancelledError) {
+        errorMessage = `Deployment canceled for ${documentPath}.`;
+      }
+      // Long-standing issue that is pretty common for all Azure calls, but can be fixed with a simple reload of VS Code.
+      // https://github.com/microsoft/vscode-azure-account/issues/53
+      else if (parseError(err).message === "Entry not found in cache.") {
+        errorMessage = `Deployment canceled for ${documentPath}. Your VS Code window must be reloaded to perform this action.`;
```

Comment:
Let's discuss wording during today's meeting.  I'd suggest something like:

Deployment failed for ${documentPath}. ADAL cache is out of date.  Please reload VS Code and try again. If this problem persists, consider changing the VS Code setting "Azure: Authentication Library" to "MSAL".
Your VS Code window must be reloaded to perform this action.

Reply:
As discussed offline, changing it to:
Deployment failed for ${documentPath}. Token cache is out of date. Please reload VS Code and try again. If this problem persists, consider changing the VS Code setting "Azure: Authentication Library" to "MSAL".

Reply:
Updated here:
https://github.com/Azure/bicep/commit/37a05b47850dc0a3c814874d459307d5bcc23fcd